src/blocks/MapApp.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect, useRef } from 'react';
+import Block from './Block';
+import { BlockModel } from './types';
+import BlockFactory from './BlockFactory';
+import { Geolocation } from '@capacitor/geolocation';
+import './BlockStyles.css';
+import { Search } from 'react-feather';
+
+// Import Google Maps type declarations
+/// <reference path="./types/google-maps.d.ts" />
+
+interface LatLngLiteral {
+  lat: number;
+  lng: number;
+}
+
+const isSuperApp = process.env.REACT_APP_IS_SUPERAPP === 'true';
```

Comment:
this isn't a sustainable pattern for miniapp developers -- they shouldn't need to conditionally determine if they are in the superapp. instead, we should store the api key in the `env` and make sure that the api is setup here as necessary.

additionally, we don't want to add another superapp-level script in the `index.tsx` to setup a miniapp api -- we should only set it up when it's needed, which is only in the edit mode of the MapApp.

Reply:
gotcha yeah i think just letting people provide their own API key as you suggested via DM is the way to go gonna fix this rn